flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobExceptionsInfo.java

```diff
@@ -156,4 +182,56 @@ public int hashCode() {
             return Objects.hash(timestamp, exception, task, location);
         }
     }
+
+    /** Nested class to hold exception information of previous failures. */
+    public static final class JobExceptionHistoryEntry {
```

Comment:
we could move this to the upper level, and have JobExceptionsInfo extend this class; then we wouldn't duplicate the entire thing.